src/App.tsx

```diff
@@ -1,223 +1,252 @@
+    setActiveTodosIds(completedTodos.map(todo => todo.id));
+
+    try {
+      await Promise.all(completedTodos.map(async todo => {
+        await deleteTodo(todo.id);
+      }));
+    } catch (error) {
+      setNotification(Notifications.Delete);
+
+      setTimeout(() => {
+        setNotification(Notifications.None);
+      }, 3000);
+    }
+
+    setActiveTodosIds([]);
+  };
+
+  const togleStatus = async (
+    id: number,
+    completed: boolean,
+    isLoading: (value: boolean) => void,
+  ) => {
+    setActiveTodosIds(todosIds => [...todosIds, id]);
+
+    try {
+      isLoading(true);
+      await updateTodoCompleted(id, !completed);
+    } catch (error) {
+      setNotification(Notifications.Update);
+      isLoading(false);
+
+      setTimeout(() => {
+        setNotification(Notifications.None);
+      }, 3000);
+    }
+
+    isLoading(false);
+    setActiveTodosIds(todosIds => todosIds.filter(todosId => todosId !== id));
+  };
+
+  const activeTodos = filterTodos(Filter.Active);
+
+  const toggleAllTodos = async () => {
+    const activeIds = activeTodos.map(todo => todo.id);
+    const allIds = todos.map(todo => todo.id);
+
+    try {
+      setActiveTodosIds(todosIds => [...todosIds, ...activeIds]);
+      await Promise.all(activeTodos
+        .map(todo => updateTodoCompleted(todo.id, true)));
+      setActiveTodosIds([]);
+
+      if (completedTodos.length === todos.length) {
+        setActiveTodosIds(todosIds => [...todosIds, ...allIds]);
+        await Promise.all(todos
+          .map(todo => updateTodoCompleted(todo.id, !todo.completed)));
+        setActiveTodosIds([]);
+      }
+    } catch {
+      setNotification(Notifications.Update);
+      setActiveTodosIds([]);
+
+      setTimeout(() => {
+        setNotification(Notifications.None);
+      }, 3000);
+    }
+  };
+
+  const filteredTodos = filterTodos(filter);
 
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
         <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
-
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
-            />
-          </form>
-        </header>
-
-        <section className="todoapp__main" data-cy="TodoList">
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                defaultChecked
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">HTML</span>
+          {!!filteredTodos.length && (
```

Comment:
```suggestion
          {filteredTodos.length > 0 && (
```
https://mate-academy.github.io/style-guides/javascript.html#153use-shortcuts-for-booleans-but-explicit-comparisons-for-strings-and-numbers